driver-core/src/main/com/mongodb/internal/operation/AsyncQueryBatchCursor.java

```diff
@@ -373,6 +377,11 @@ private void handleGetMoreQueryResult(final AsyncConnection connection, final Si
         }
     }
 
+    private void logQueryResult(final QueryResult<T> result) {
+        LOGGER.debug(format("Received batch of %d documents with cursorId %d from server %s", result.getResults().size(),
```

Comment:
Suggestion: Add an if statement to see if debug is enabled before composing the string and passing it to debug.

Reply:
I considered that, but generally we don't do that for cases where the formatting is relatively cheap, as it is here, and where the messages are only logged once per I/O activity, as it is here.  The cost is far exceeded by the I/O.